extensions/wikia/WallNotifications/WallNotifications.class.php

```diff
@@ -387,50 +387,65 @@ protected function sendEmails( array $watchers, WallNotificationEntity $notifica
 				( $mode == WALL_EMAIL_SINCEVISITED && empty( $this->uniqueUsers[$entityKey][$watcher->getId()] ) )
 			)) {
 
-				$key = $this->createKeyForMailNotification( $watcher->getId(), $notification );
-				$watcherName = $watcher->getName();
-
-				if ( $notification->data->msg_author_username == $notification->data->msg_author_displayname ) {
-					$author_signature = $notification->data->msg_author_username;
+				$controller = $this->getEmailExtensionController( $notification );
+				if ( !empty( $controller ) ) {
+					$params = [
+						'boardNamespace' => $notification->data->article_title_ns,
+						'boardTitle' => $notification->data->article_title_text,
+						'titleText' => $notification->data->thread_title,
+						'titleUrl' => $notification->data->thread_title,
+						'details' => $text,
+						'targetUser' => $watcher->getName(),
+						'fromAddress' => $this->app->wg->PasswordSender,
+						'replyToAddress' => $this->app->wg->NoReplyAddress,
+						'fromName' => $notification->data->msg_author_username
+					];
+					F::app()->sendRequest( $controller, 'handle', $params );
 				} else {
-					$author_signature = $notification->data->msg_author_displayname .
-						' (' . $notification->data->msg_author_username . ')';
-				}
+					$key = $this->createKeyForMailNotification( $watcher->getId(), $notification );
+					$watcherName = $watcher->getName();
 
-				$data = [];
-				wfRunHooks( 'NotificationGetMailNotificationMessage', [
-					&$notification, &$data, $key, $watcherName, $author_signature, $textNoHtml, $text
-				]);
-				if ( empty( $data ) ) {
-					$data = [
-						'$WATCHER' => $watcherName,
-						'$WIKI' => $notification->data->wikiname,
-						'$PARENT_AUTHOR_NAME' => 	(empty($notification->data->parent_displayname)
-													? ''
-													: $notification->data->parent_displayname),
-						'$AUTHOR_NAME' => $notification->data->msg_author_displayname,
-						'$AUTHOR' => $notification->data->msg_author_username,
-						'$AUTHOR_SIGNATURE' => $author_signature,
-						'$MAIL_SUBJECT' => wfMessage('mail-notification-subject', [
-							'$1' => $notification->data->thread_title,
-							'$2' => $notification->data->wikiname
-						])->text(),
-						'$METATITLE' => $notification->data->thread_title,
-						'$MESSAGE_LINK' =>  $notification->data->url,
-						'$MESSAGE_NO_HTML' =>  $textNoHtml,
-						'$MESSAGE_HTML' =>  $text,
-						'$MSG_KEY_SUBJECT' => $key,
-						'$MSG_KEY_BODY' => 'mail-notification-body',
-						'$MSG_KEY_GREETING' => 'mail-notification-html-greeting',
-					];
-				}
+					if ( $notification->data->msg_author_username == $notification->data->msg_author_displayname ) {
+						$author_signature = $notification->data->msg_author_username;
+					} else {
+						$author_signature = $notification->data->msg_author_displayname .
+							' (' . $notification->data->msg_author_username . ')';
+					}
+
+					$data = [];
+					wfRunHooks( 'NotificationGetMailNotificationMessage', [
```

Comment:
The Forum extension is using this hook, but we're sending the forum emails up above. Do you know if this hook is still needed? If so, we should move it up around L398, otherwise we should probably delete that hook.